### PR TITLE
Fastlane TestFlight Support

### LINKED ISSUE #12: Add support for automatic TestFlight deployments
### Depends on #6 

Add a CI workflow that automatically deploys builds to TestFlight.

### PR DESCRIPTION
**Pull request by H1ghBre4k3r**
Add support for TestFlight deployments via Fastlane. 
 - action gets triggered on tag pushes
 - extracts version number from tag
 - sets xcode workspace up
 - downloads certs via `match`
 - sets app version and upgrades build number
 - builds and signs the app
 - uploads app to TestFlight
 
 Closes #12

### CODE REVIEW

File: .github/workflows/testflight.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Testflight
+
+on:
+  push:
+    tags:
+      - 'v*.*.*-beta*'
+
+jobs:
+  testflight:
+    runs-on: macos-latest
```
Comment (fwcd) [inline .github/workflows/testflight.yml:10]: Looks good, just an idea: Perhaps we could move this job into the iOS workflow along with a workflow dependency on `build` to reuse the setup and (ideally) even the build products. Not a blocker though.

File: .gitignore [hunk 2]
```diff
@@ -65,3 +65,5 @@ buck-out/
 
 # Typedoc
 /docs
+
+.env*
```
Comment (fwcd) [inline .gitignore:69]: Minor style nitpick: Trailing newlines can be inserted automatically on save with

```json
"files.insertFinalNewline": true
```

in VSCode, they play better with command-line tools like `cat` or `wc`.

**Review by fwcd — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (H1ghBre4k3r) [reply .gitignore]: Inserted trailing newlines in files where they were missing.

File: .github/workflows/testflight.yml (hunk 1 above)
Comment (H1ghBre4k3r) [reply .github/workflows/testflight.yml:10]: Great idea! Added #40 for that.

**Pull request merged**